packages/local_auth/local_auth_ios/ios/Classes/FLTLocalAuthPlugin.m

```diff
@@ -212,54 +205,68 @@ - (void)handleAuthReplyWithSuccess:(BOOL)success
+                                                         resultHandler:completion];
         } else {
-          result(@NO);
+          [self handleSucceeded:NO withCompletion:completion];
         }
         return;
     }
-    [self handleErrors:error flutterArguments:arguments withFlutterResult:result];
+    [self handleError:error withOptions:options strings:strings completion:completion];
   }
 }
 
-- (void)handleErrors:(NSError *)authError
-     flutterArguments:(NSDictionary *)arguments
-    withFlutterResult:(FlutterResult)result {
-  NSString *errorCode = @"NotAvailable";
+- (void)handleSucceeded:(BOOL)succeeded withCompletion:(nonnull AuthCompletion)completion {
+  completion(
+      [FLAAuthResultDetails makeWithValue:(succeeded ? FLAAuthResultSuccess : FLAAuthResultFailure)
+                             errorMessage:nil
+                             errorDetails:nil],
+      nil);
+}
+
+- (void)handleError:(NSError *)authError
+        withOptions:(FLAAuthOptions *)options
+            strings:(FLAAuthStrings *)strings
+         completion:(nonnull AuthCompletion)completion {
+  FLAAuthResult result = FLAAuthResultErrorNotAvailable;
   switch (authError.code) {
     case LAErrorPasscodeNotSet:
     case LAErrorBiometryNotEnrolled:
-      if ([arguments[@"useErrorDialogs"] boolValue]) {
-        [self alertMessage:arguments[@"goToSettingDescriptionIOS"]
-                 firstButton:arguments[@"okButton"]
-               flutterResult:result
-            additionalButton:arguments[@"goToSetting"]];
+      if (options.useErrorDialgs.boolValue) {
+        [self showAlertWithMessage:strings.goToSettingsDescription
+                       firstButton:strings.cancelButton
+                  additionalButton:strings.goToSettingsButton
+                        completion:completion];
         return;
       }
-      errorCode = authError.code == LAErrorPasscodeNotSet ? @"PasscodeNotSet" : @"NotEnrolled";
+      result = authError.code == LAErrorPasscodeNotSet ? FLAAuthResultErrorPasscodeNotSet
+                                                       : FLAAuthResultErrorNotEnrolled;
       break;
     case LAErrorBiometryLockout:
-      [self alertMessage:arguments[@"lockOut"]
-               firstButton:arguments[@"okButton"]
-             flutterResult:result
-          additionalButton:nil];
+      [self showAlertWithMessage:strings.lockOut
+                     firstButton:strings.cancelButton
+                additionalButton:nil
+                      completion:completion];
       return;
   }
-  result([FlutterError errorWithCode:errorCode
-                             message:authError.localizedDescription
-                             details:authError.domain]);
+  completion([FLAAuthResultDetails makeWithValue:result
+                                    errorMessage:authError.localizedDescription
+                                    errorDetails:authError.domain],
+             nil);
```

Comment:
is it more natural to just complete with an NSError here? Or is there any issue for pigeon to handle error case properly? 

```
completion(result, [NSError errorWith...])
```

Reply:
oops FlutterError rather.